cmd/setup.go

```diff
@@ -0,0 +1,214 @@
+// Copyright © 2018 NAME HERE <EMAIL ADDRESS>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/BurntSushi/toml"
+	"github.com/netm4ul/netm4ul/core/database"
+	"github.com/spf13/cobra"
+
+	mgo "gopkg.in/mgo.v2"
+)
+
+const (
+	defaultDBSetupUser     = "admin"
+	defaultDBSetupPassword = "admin"
+	dbname                 = "NetM4ul"
+)
+
+var (
+	cliDBSetupUser     string
+	cliDBSetupPassword string
+	userIMode          = false
+	passwordIMode      = false
+)
+
+// setupCmd represents the setup command
+var setupCmd = &cobra.Command{
+	Use:   "setup",
+	Short: "NetM4ul setup",
+	Long: `A longer description that spans multiple lines and likely contains examples
+and usage of using your command. For example:
+
+Cobra is a CLI library for Go that empowers applications.
+This application is a tool to generate the needed files
+to quickly create a Cobra application.`,
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		createSessionBase()
+		CLISession.Config.Database.User = cliDBSetupUser
+		CLISession.Config.Database.Password = cliDBSetupPassword
+	},
+
+	Run: func(cmd *cobra.Command, args []string) {
+		ex2Conf()
+		if CLISession.Config.Database.User == "" {
+			userIMode = true
+		}
+		if CLISession.Config.Database.Password == "" {
+			passwordIMode = true
+		}
+		setupDB()
+	},
+}
+
+func check(err error) {
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// prompt user for configuration parameters
+func prompt(param string) (answer string) {
+	var text string
+	var input string
+	var defInput string
+
+	// Database parameters
+	if param == "dbuser" {
+		defInput = defaultDBSetupUser
+		fmt.Println("Interactive mode for username (default : ", defInput, ")")
+		text = "Enter username: "
+	} else if param == "dbpassword" {
+		defInput = defaultDBSetupPassword
+		fmt.Println("Interactive mode for password (default : ", defInput, ")")
+		text = "Enter password: "
+	}
+
```

Comment:
All of this if / else if should be replaced with something like :
 
```golang
type PromptRes struct{
    Message string
    DefaultValue string
}

promptString := map[string]PromptRes{
    "dbuser" : PromptRes{Message : "Interactive mode for username (default : %s)", DefaultValue : defaultDBSetupUser},
    "dbpassword" : PromptRes{Message : "Interactive mode for password (default : %s)", DefaultValue : defaultDBSetupPassword},
}

fmt.Printf(promptString[param].Message, promptString[param].DefaultValue)
```

(and remove the following comments)